rundeckapp/grails-app/services/rundeck/services/NotificationService.groovy

```diff
@@ -493,7 +493,20 @@ public class NotificationService implements ApplicationContextAware{
                     execMap.context=context
                     Map config= n.configuration
                     if (context && config) {
-                        config = DataContextUtils.replaceDataReferences(config, context)
+                        //adding the execution variables in order to apply a replacement of config data (for example replace things like ${execution.xxx} on config)
+
+                        //clone the original execMap to exclude java list and numeric values
+                        def newExecMap = execMap.clone()
```

Comment:
I think this same behavior is already done for email notifications here: https://github.com/rundeck/rundeck/pull/5689/files#diff-7c0af42b7418a4cfa6558c80b306ec26R243-R253

is there a reason to do it differently?

Reply:
I will check it